src/components/ParserComponent.tsx

```diff
@@ -290,8 +324,14 @@ export default class ParserComponent extends React.Component<ParserInterface,
           }
         }
         );
+        // if there is data then arrayOfColumns shouldn't be empty
+        assert.notStrictEqual(arrayOfColumns, [],
+            'inferTypes(): arrayOfColumns is empty after parsing data');
         return arrayOfColumns;
       } else {
+        assert(this.state.data.length === 0,
+            'inferTypes(): this.state.data is not empty');
+        // should probably return undefined here?
```

Comment:
No need, since we throw an exception there's no return value (unless we catch it in this function too